pkg/resource/public_service.go

```diff
@@ -48,7 +48,15 @@ func (b PublicServiceBuilder) Build(obj client.Object) error {
 		service.ObjectMeta.Labels = map[string]string{}
 	}
 
-	service.Annotations = b.Spec().AdditionalAnnotations
+	annotations :=  make(map[string]string)
+	for k,v := range  b.Spec().AdditionalAnnotations {
+		annotations[k] = v
+	}
+	for k,v := range  b.Spec().PublicServiceAnnotations {
+		annotations[k] = v
+	}
```

Comment:
the two for loops.. probably not the best way to handle it. I tried a few other ways but if AdditionalAnnotations wasn't set it kept erroring out for a nil map. Open to ideas on how to make this better